File: apps/mobile-portfolio/src/types/userPreference.ts
export interface UserPreference {
  preference_id: string;
  customer_id: string;
  theme: string | null;
  language: string | null;
  notifications_enabled: boolean;
  updated_at: string;
}

export interface UserPreferenceCreate {
  customer_id: string;
  theme?: string | null;
  language?: string | null;
  notifications_enabled?: boolean;
}

File: apps/server-go/internal/domain/user_preference.go
package domain

import "time"

type UserPreference struct {
	PreferenceID        string    `json:"preference_id"`
	CustomerID          string    `json:"customer_id"`
	Theme               *string   `json:"theme,omitempty"`
	Language            *string   `json:"language,omitempty"`
	NotificationsEnabled bool     `json:"notifications_enabled"`
	UpdatedAt           time.Time `json:"updated_at"`
}
